src/FontManager.ts

```diff
@@ -141,17 +152,21 @@ export class FontManager extends EventDispatcher<EventMap> {
         // don't end up with *too* much space between characters.
         xAdvance: Math.max(sdf.glyphAdvance, sdf.width - buffer),
       };
-      maxAscent = Math.max(maxAscent, sdf.glyphTop);
-      x += sdf.width;
+      this.atlasData.maxAscent = Math.max(this.atlasData.maxAscent, sdf.glyphTop);
+      this.#lastCharPosition.x += sdf.width;
+    }
+
+    for (const char of newAlphabet) {
+      this.alphabet += char;
     }
 
     this.atlasData = {
       data: atlas,
       width: atlasWidth,
       height: atlasHeight,
       charInfo,
```

Comment:
If these 4 props are never changing, I am questioning whether we need to recreate the atlasData object at all? Maybe it's time for it to stop being a separate object and just store the Uint8ClampedArray directly on the FontManager -- I don't remember historically how we ended up here (maybe the atlas size used to change more at some point) but it does seem like it could be simplified now.

Reply:
it's fine to change the interface 👍